test/classes.coffee

```diff
@@ -1808,3 +1808,19 @@ test 'Bound method of immediately instantiated class with expression base class
 
   {derivedBound} = a
   eq derivedBound(), 3
+
+test "#4591: super.x.y, super['x'].y", ->
```

Comment:
well, `super.foo` still has the implicit call there – it means `super(arguments...).foo`